h/auth/policy.py

```diff
@@ -78,6 +150,57 @@ def forget(self, request):
         """Not implemented for basic auth client policy."""
         return []
 
+    @classmethod
+    def check(cls, username, password, request):
```

Comment:
I will change them to static.

I concur that `_forwarded_user_id` is unambiguously a helper function, but I think there is a (small, admittedly) argument that `check` could be public; it feels quite standalone in my tests.

That is, you can do `AuthClientPolicy.check(request)` on any ol' Pyramid request without instantiating an instance of the policy and it will return the correct principals (or `None`).